### PR TITLE
[CSR-1773] feat: Add rules example for github actions

### PR DESCRIPTION
**Pull request by twk3**
Updates config with support for currents fixtures

- I did fix the file-upload tests in this PR, but left the other existing spec failures.

- The rules and failure specs always should fail (rules will be fixed after we have the UI in prod)
- accessibility specs always fail (existing), it's doing a test against currents.dev as an example, and we don't pass
- pom specs always fail (existing), these test a remote site, and haven't been updated to pass in quite some time

### CODE REVIEW

File: .github/test-basic-pwc-failed-only.yml [hunk 1]
```diff
@@ -30,7 +30,7 @@ jobs:
     name: "Playwright Tests - check failed"
     timeout-minutes: 60
     runs-on: ubuntu-22.04
-    container: mcr.microsoft.com/playwright:latest
+    container: mcr.microsoft.com/playwright:v1.49.0-jammy
```
Comment (twk3) [inline .github/test-basic-pwc-failed-only.yml:33]: the `latest` tag doesn't seem to get updated anymore

File: basic/pw.config.shared.ts [hunk 2]
```diff
@@ -37,8 +39,16 @@ const config: PlaywrightTestConfig = {
     },
   ],
 
+  webServer: {
```
Comment (twk3) [inline basic/pw.config.shared.ts:42]: This should get the existing file-upload spec test working.

File: .github/test-basic-pwc-failed-only.yml (hunk 1 above)
Comment (ynahmany) [reply .github/test-basic-pwc-failed-only.yml:33]: yes, did the same on currents-playwright repo.

File: basic/1-getting-started.spec.ts [hunk 3]
```diff
@@ -1,4 +1,5 @@
-import { expect, test } from "@playwright/test";
+import { expect } from "@playwright/test";
+import { test } from "./base.ts";
 
 test("basic test @basic", async ({ page }) => {
   await page.goto("https://todomvc.com/examples/backbone/dist/");
```
Comment (ynahmany) [inline basic/1-getting-started.spec.ts:5]: they changed path - so this one will work
https://todomvc.com/examples/javascript-es6/dist

File: basic/8-rules.spec.ts [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+ * All tests in this file are matched with a Currents Quarantine Rule based on filename.
+ * See the `rules` fixture in `base.ts` to see how the rules fixtures were configured,
+ * and the docs for more information on how to configure rules: https://docs.currents.dev/guides/automation-rules
+ */
+
+/**
+ * This is a failing test that will be quarantined by the file based rule
+ */
+test("rules quarantine test", async ({ page }) => {
+  await page.goto("https://todomvc.com/examples/backbone/dist/");
```
Comment (ynahmany) [inline basic/8-rules.spec.ts:14]: it is 404 page.
https://todomvc.com/examples/javascript-es6/dist
this one should work

File: basic/8-rules.spec.ts [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+  await inputBox.fill("Learn JavaScript");
+  await inputBox.press("Enter");
+  await expect(todoList).toHaveText("Learn Playwright");
+});
+
+/**
+ * This tests will be completely skipped by a rule that matches fileName + test name
+ */
+test("rules skip test", async ({ page }) => {
+  await page.goto("https://todomvc.com/examples/backbone/dist/");
```
Comment (ynahmany) [inline basic/8-rules.spec.ts:29]: it is 404 page.
https://todomvc.com/examples/javascript-es6/dist
this one should work

**Review by ynahmany — CHANGES_REQUESTED** [review]
The notes are not mandatory - i can change it and make sure this repo tests is up to date if you like.

**Comment by twk3** [comment]
@ynahmany this should be ready for merge. I will add the rules to prod once we deploy to prod, but I added details on what those rules would be in the comments on the file here.

**Pull request merged**